keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1535 @@
+
+A device with `allowMultipleAllocations` property can only be allocated
+when its consumability has been verified and its attributes match the request's selectors and constraints.
+The newly added `consumedCapacities` field in the `DeviceRequestAllocationResult` will be set to the calculated capacity upon a successful allocation.
+This value may differ from the originally requested amount, as it is rounded up to the nearest valid value based on the device’s sharing policy.
+If no specific amount is requested, a default consumption value is applied.
+
+### API enhancement
+To enable this enhancement, the following API updates are proposed.
+
+#### ResourceSliceSpec's Device
+
+```go
+
+// Device represents one individual hardware instance that can be selected based
+// on its attributes. Besides the name, exactly one field must be set.
+type Device struct {
+...
+   // AllowMultipleAllocations marks whether the device is allowed to be allocated for multiple times.
+   //
+   // A device with allowMultipleAllocations="true" can be allocated more than once,
+   // and its capacity is shared, regardless of whether the CapacitySharingPolicy is defined or not.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   AllowMultipleAllocations *bool
+}
+
+// DeviceCapacity describes a quantity associated with a device.
+type DeviceCapacity struct {
+    // Value defines how much of a certain device capacity is available.
+    //
+    // If the capacity is consumable (i.e., a ClaimPolicy is specified),
+    // the consumed amount is deducted and cached in memory by the scheduler.
+    // Note that the remaining capacity is not reflected in the resource slice.
+    //
+    // +required
+    Value resource.Quantity
+
+   // SharingPolicy specifies that this device capacity must be consumed
+   // by each resource claim according to the defined sharing policy.
+   // The Device must allow multiple allocations.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   SharingPolicy *CapacitySharingPolicy
+}
+
+// CapacitySharingPolicy defines how requests consume the available capacity.
+// A policy must have a default value to be applied when no value is explicitly provided.
+// It can either specify a range of valid values or a discrete set of them.
+// Exactly one of them must be defined.
+// The default value must be a valid value.
+type CapacitySharingPolicy struct {
+   // Default specifies the default capacity to be used for a consumption request.
+   //
+   // +required
+   Default resource.Quantity
+
+   // DiscreteValues defines a set of acceptable quantity values in consuming requests.
+   //
+   // +optional
+   // +oneOf=ValidSharingValues
+   DiscreteValues *CapacitySharingPolicyDiscrete
+
+   // ValueRange defines an acceptable quantity value range in consuming requests.
+   //
+   // +optional
+   // +oneOf=ValidSharingValues
+   ValueRange *CapacitySharingPolicyRange
```

Comment:
non-blocking, can think about this as we implement: if we rename the above option to `ValidValues`, this could rename to `ValidRange` or something similar